packages/front-end/pages/metrics.tsx

```diff
@@ -19,6 +19,56 @@ import TagsFilter, {
   useTagsFilter,
 } from "../components/Tags/TagsFilter";
 import SortedTags from "../components/Tags/SortedTags";
+import { isNullUndefinedOrEmpty } from "../services/utils";
+
+interface MetricToolTipCompProps {
+  metric: MetricInterface;
+}
+
+function truncateMetricDescription(metricDescription: string) {
+  if (metricDescription.length < 300) return metricDescription;
+  return `${metricDescription.substring(0, 300)}...`;
+}
+
+const MetricToolTipComp = ({
+  metric,
+}: MetricToolTipCompProps): React.ReactElement => {
+  return (
+    <div className="text-left">
+      {!isNullUndefinedOrEmpty(metric.description) && (
+        <div>
+          <b>Description:</b> {truncateMetricDescription(metric.description)}
+        </div>
+      )}
+      {!isNullUndefinedOrEmpty(metric.type) && (
+        <div>
+          <b>Type:</b> {metric.type}
+        </div>
+      )}
+      {!isNullUndefinedOrEmpty(metric.tags) && metric.tags.length > 0 && (
+        <div>
+          <b>Tags:</b>{" "}
+          <SortedTags tags={Object.values(metric.tags)} color="purple" />
+        </div>
+      )}
+      {!isNullUndefinedOrEmpty(metric.cap) && (
+        <div>
+          <b>Cap:</b> {metric.cap}
+        </div>
+      )}
+      {!isNullUndefinedOrEmpty(metric.conversionDelayHours) && (
+        <div>
+          <b>Conversion Delay Hours:</b> {metric.conversionDelayHours}
+        </div>
+      )}
+      {!isNullUndefinedOrEmpty(metric.conversionWindowHours) && (
+        <div>
+          <b>Conversion Window Hours:</b> {metric.conversionWindowHours}
+        </div>
+      )}
```

Comment:
There's probably a more elegant way to do this that doesn't involve as much repetitive code.

Reply:
I did a refactor to this. Checkout MetricTooltipBodyRow.tsx and let me know your thoughts.